source/Messages.cpp

```diff
@@ -88,5 +105,28 @@ void Messages::Reset()
 {
 	lock_guard<mutex> lock(incomingMutex);
 	incoming.clear();
-	list.clear();
+	recent.clear();
+	log.clear();
+}
+
+
+
+// Get color that should be used for drawing messages of given importance.
+const Color *Messages::GetColor(Importance importance, bool isLog)
+{
+	string prefix = (isLog ? "message log importance " : "message importance ");
+	switch(importance)
+	{
+		case Messages::Importance::Highest:
+			return GameData::Colors().Get(prefix + "highest");
+		case Messages::Importance::High:
+			return GameData::Colors().Get(prefix + "high");
+		case Messages::Importance::Info:
+			return GameData::Colors().Get(prefix + "info");
+		case Messages::Importance::Daily:
+			return GameData::Colors().Get(prefix + "daily");
+		case Messages::Importance::Low:
+			return GameData::Colors().Get(prefix + "low");
+	}
+	assert(false && "invalid message importance");
```

Comment:
why not return a default color and spit out a warning?

Reply:
Producing a warning or error here wouldn't be very useful.
Those are generally related to game data issues or something specific to the environment the game is running in (audio problems, for example.)
If there isn't a case here for the value of `importance`, there's not a whole lot an end user can do about it. Fixing that requires editing the source code and recompiling the game executable.
With this assertion, such a failure will produce a runtime error, which I also think is inappropriate.
(And, as it is, this method seems to be causing a failure to compile anyway.)
I would probably put the "Importance::LOW" case outside the switch, or put the default line immediately above it.